src/ecs/commands/runCommandInContainer.ts

```diff
@@ -13,14 +13,22 @@ import { ChildProcess } from '../../shared/utilities/childProcess'
 import { EcsContainerNode } from '../explorer/ecsContainerNode'
 import { recordEcsRunExecuteCommand } from '../../shared/telemetry/telemetry.gen'
 import { DefaultSettingsConfiguration, SettingsConfiguration } from '../../shared/settingsConfiguration'
-import { extensionSettingsPrefix, INSIGHTS_TIMESTAMP_FORMAT } from '../../shared/constants'
+import { ecsRequiredPermissionsUrl, extensionSettingsPrefix, INSIGHTS_TIMESTAMP_FORMAT } from '../../shared/constants'
 import { showOutputMessage, showViewLogsMessage } from '../../shared/utilities/messages'
 import { getOrInstallCli } from '../../shared/utilities/cliUtils'
 import { removeAnsi } from '../../shared/utilities/textUtilities'
 import globals from '../../shared/extensionGlobals'
 import { CommandWizard } from '../wizards/executeCommand'
 import { CancellationError } from '../../shared/utilities/timeoutUtils'
 
+// Required SSM permissions for the task IAM role, see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html#ecs-exec-enabling-and-using
+const REQUIRED_SSM_PERMISSIONS = [
+    'ssmmessages:CreateControlChannel',
+    'ssmmessages:CreateDataChannel',
+    'ssmmessages:OpenControlChannel',
+    'ssmmessages:OpenDataChannel',
```

Comment:
Are these the correct persmissions? I created a role with readonly permissions, tried to execute a command (with these changes)

> Failed to execute command in container, [AccessDeniedException: User: arn:aws:sts::111:assumed-role/cloud9user/foo is not authorized to perform: ecs:ExecuteCommand on resource: arn:aws:ecs:us-west-2:111:cluster/jan7 because no identity-based policy allows the ecs:ExecuteCommand action

Reply:
The user's credentials are not what are being tested. It's the role defined in the task definition that needs the SSM permissions for ECS Exec. A new PR will be created to add verification for the user's ECS permissions.